src/Model/Event.php

```diff
@@ -577,75 +651,84 @@ public static function prepareListForTemplate(array $event_result): array
+		$day   = DI::l10n()->getDay($day);
 
-			// Show edit and drop actions only if the user is the owner of the event and the event
-			// is a real event (no bithdays).
-			$edit = null;
-			$copy = null;
-			$drop = null;
-			if (DI::userSession()->getLocalUserId() && DI::userSession()->getLocalUserId() == $event['uid'] && $event['type'] == 'event') {
-				$edit = !$event['cid'] ? [DI::baseUrl() . '/events/event/' . $event['id'], DI::l10n()->t('Edit event')     , '', ''] : null;
-				$copy = !$event['cid'] ? [DI::baseUrl() . '/events/copy/' . $event['id'] , DI::l10n()->t('Duplicate event'), '', ''] : null;
-				$drop =                  [DI::baseUrl() . '/events/drop/' . $event['id'] , DI::l10n()->t('Delete event')   , '', ''];
-			}
+		if ($event['nofinish']) {
+			$end = null;
+		} else {
+			$end = DateTimeFormat::local($event['finish'], 'c');
+		}
 
-			$title = BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['summary']));
-			if (!$title) {
-				list($title, $_trash) = explode("<br", BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['desc'])), BBCode::API);
-			}
+		// Show edit and drop actions only if the user is the owner of the event and the event
+		// is a real event (no bithdays).
+		$edit = null;
+		$copy = null;
+		$drop = null;
+		if (DI::userSession()->getLocalUserId() && DI::userSession()->getLocalUserId() == $event['uid'] && $event['type'] == 'event') {
+			$edit = !$event['cid'] ? [DI::baseUrl() . '/calendar/event/edit/' . $event['id'], DI::l10n()->t('Edit event')     , '', ''] : null;
+			$copy = !$event['cid'] ? [DI::baseUrl() . '/calendar/event/copy/' . $event['id'] , DI::l10n()->t('Duplicate event'), '', ''] : null;
+			$drop =                  [DI::baseUrl() . '/calendar/api/delete/' . $event['id'] , DI::l10n()->t('Delete event')   , '', ''];
+		}
 
-			$author_link = $event['author-link'];
-
-			$event['author-link'] = Contact::magicLink($author_link);
-
-			$html = self::getHTML($event);
-			$event['summary']  = BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['summary']));
-			$event['desc']     = BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['desc']));
-			$event['location'] = BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['location']));
-			$event_list[] = [
-				'id'       => $event['id'],
-				'start'    => $start,
-				'end'      => $end,
-				'allDay'   => false,
-				'title'    => $title,
-				'j'        => $j,
-				'd'        => $day,
-				'edit'     => $edit,
-				'drop'     => $drop,
-				'copy'     => $copy,
-				'is_first' => $is_first,
-				'item'     => $event,
-				'html'     => $html,
-				'plink'    => Item::getPlink($event),
-			];
+		$title = BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['summary']));
+		if (!$title) {
+			[$title, $_trash] = explode("<br", BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['desc'])), BBCode::API);
```

Comment:
A less complex version of the same call, notice the `limit` parameter used to prevent any further parsing after the first `<br` encountered.
```suggestion
			$title = explode('<br', BBCode::convertForUriId($event['uri-id'], Strings::escapeHtml($event['desc'])), BBCode::API, 2)[0];
```